docs/specs/ML.Net Sweepable API.md

```diff
@@ -0,0 +1,162 @@
+sweepable estimator allows user to combine search space with estimators in a similar way how ml.net estimator/pipeline are created. You use `CreateSweepableEstimator`, which accepts a lambda function and a search space to create a sweepable estimator. And it also provides `.Append` extension method so you can append sweepable estimator similiar with how you append other ml.net extimator. The bellow example presents how to create a pipeline with two sweepable estimators, one for text featurizor and the other for fast tree, for `titanic` dataset via `Sweepable()` extension.
+
+``` csharp
+var context = new MLContext();
+var fastTreeSS = new SearchSpace<FastTreeOption>();
+var textFeaturizeSS = new SearchSpace<FeaturizeTextOption>();
+
+var pipeline = context.Transforms.Categorical.OneHotEncoding(new[] { new InputOutputColumnPair(@"Sex", @"Sex"), new InputOutputColumnPair(@"Embarked", @"Embarked") })
+                .Append(context.Transforms.Concatenate(@"TextFeature", @"Name", "Ticket", "Cabin"))
+                .Append(context.Sweepable().CreateSweepableEstimator(
+                    (mlContext, option) =>
+                    {
+                        var textOption = new TextFeaturizingEstimator.Options
+                        {
+                            CaseMode = option.CaseMode,
+                            KeepDiacritics = option.KeepDiacritics,
+                            KeepNumbers = option.KeepNumbers,
+                            KeepPunctuations = option.KeepPunctuations,
+                            CharFeatureExtractor = new WordBagEstimator.Options()
+                            {
+                                NgramLength = option.WordBagEstimatorOption.NgramLength,
+                                UseAllLengths = option.WordBagEstimatorOption.UseAllLengths,
+                                Weighting = option.WordBagEstimatorOption.WeightingCriteria,
+                            },
+                        };
+
+                        return mlContext.Transforms.Text.FeaturizeText("TextFeature", textOption);
+                    },
+                    textFeaturizeSS))
+                .Append(context.Transforms.Concatenate(@"Features", new[] { @"Sex", @"Embarked", @"Pclass", @"Age", @"SibSp", @"Parch", @"Fare", "TextFeature" }))
+                .Append(context.Transforms.Conversion.ConvertType("Survived", "Survived", Data.DataKind.Boolean))
+                .Append(context.Sweepable().CreateSweepableEstimator(
+                    (mlContext, option) => mlContext.BinaryClassification.Trainers.FastForest(labelColumnName: "Survived", featureColumnName: "Features", numberOfLeaves: option.NumberOfLeavenumberOfTrees: option.NumberOfTrees),
+                    fastTreeSS))
+                .Append(context.BinaryClassification.Calibrators.Naive(labelColumnName: @"Survived", scoreColumnName: @"Score"));
+```
+
+After sweepable pipeline is created, one can call `BuildTrainingPipeline` to convert it to a ml.net pipeline.

```

Comment:
Will the type of the pipeline be the same as a normal ML.NET pipeline? What happens if someone tries to run it without calling that method first?

Reply:
No it won't be the same ML.Net pipeline. And user needs to call `BuildTrainingPipeline` with passed parameter to convert it to an ML.Net pipeline and only after that, they can use `Fit/Transform` to train their model.

I feel like that `BuildTrainingPipeline` might be too detailed an API to be exposed. And it will be better if it can be wrapped and only exposes a `Run` method. We can discuss it later though.